ta4j-core/src/main/java/org/ta4j/core/BaseTradingRecord.java

```diff
@@ -133,7 +151,24 @@ public final boolean exit(int index, Decimal price, Decimal amount) {
     
     @Override
     public List<Trade> getTrades() {
-        return trades;
+    	if (trades.size() == 0) {
+    		return trades;
+    	}
+        return getTrades(
+        		trades.get(0).getEntry().getIndex(),
+        		trades.get(trades.size()-1).getExit().getIndex());
```

Comment:
Whats the difference between returning this.trades and returning all of this.trades that are between this.trades.first entry and this.trades.last entry?

Reply:
I needed a getTrades(int, int) for
```
NumberOfTradesCriterion#calculate(TimeSeries, TradingRecord, int, int)
```
I implemented getTrades(int, int) and rewrote BaseTradingRecord#getTrades() to call it for re-use.  But I changed all this so now getTrades() is a default interface method
```
default List<Trade> getTrades() {
    return getTrades(getFirstOrder().getIndex(), getLastOrder().getIndex());
}
```
and added the interface method getFirstOrder() and implemented it in baseTradingRecord.  Finally, I removed getTrades() from BaseTradingRecord.  This looks a lot better now.

I could do the same with getTradeCount() and getTradeCount(int, int) but am trying to limit my impact.